sdk/core/amqp-common/src/auth/aad.ts

```diff
@@ -1,28 +1,30 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-import {
-  ApplicationTokenCredentials,
-  DeviceTokenCredentials,
-  UserTokenCredentials,
-  MSITokenCredentials
-} from "@azure/ms-rest-nodeauth";
 import { TokenInfo, TokenType, TokenProvider } from "./token";
 import * as Constants from "../util/constants";
 
 /**
  * Defines the AAD (Azure ActiveDirectory) TokenProvider.
+ * This accepts the following credentials from the `@azure/ms-rest-nodeauth` package:
+ *  - ApplicationTokenCredentials
+ *  - UserTokenCredentials
+ *  - DeviceTokenCredentials
+ *  - MSITokenCredentials (MSIVmTokenCredentials)
  * @class AadTokenProvider
  */
 export class AadTokenProvider implements TokenProvider {
   /**
    * @property {(ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials)} credentials - The credentials object after successful authentication with AAD.
+   * This property matches what was passed into the constructor.
    */
-  credentials:
-    | ApplicationTokenCredentials
-    | UserTokenCredentials
-    | DeviceTokenCredentials
-    | MSITokenCredentials;
+  credentials: {
+    getToken(): Promise<{
+      tokenType: string;
+      accessToken: string;
+      [x: string]: any;
+    }>;
+  };
```

Comment:
Does this have to be a public property?

Reply:
I did that to match the MSIVmTokenCredential TokenResponse, but it doesn't have to be. I will have to add `expiresOn?: Date | string;` though as a result since our code depends on that field.

Reply:
Updated! I have tested these changes with ms-rest-nodeauth 3.0.3 and confirmed they work. CI tests with ms-rest-nodeauth 0.9.x to ensure that keeps working.

Reply:
Sorry, I think I misunderstood what you meant by `this` as being the index signature property on the interface. No, I don't think `credentials` needs to be a public property.

Reply:
I've not made `credentials` a private property. This didn't create any compilation errors. I also ran a test application using `ApplicationTokenCredentials` with this change and was able to send and receive messages to a queue still. I couldn't find anywhere in service-bus/amqp-common where this field would be used outside of this class.